### DISCUSSION TITLE
backoff rules  , mx grouping , scale question

### DISCUSSION THREAD

**Discussion by z3d3m0n** [postalserver/postal#3210] [Help with using Postal] [answered]
Hello,

hope everyone is safe
i have a question to postalhq gurus
i am a long time powermta user , but lack of good interface and db connection always irritated me.
the postalhq looks very good and i would like to try it for one of the b2b corporate email project.
but i have a few questions.

is there any feature to create a backoff rule to throttle down the speed or limit the connection if i get some error messages from provider and i want to give them a time out for some time

is there a feature to goup mx servers and to send them throttled messages to avoid any blocking and to follow their guideline and not to exceed the amount of emails sent from one ip 

in terms of scaling how many emails per minute i can send from a 4 CUP and 4 RAM , SSD server ? 

thx in advance

**Comment by willpower232** [comment]
> is there any feature to create a backoff rule to throttle down the speed or limit the connection if i get some error messages from provider and i want to give them a time out for some time

Postal will block sending to individual addresses if they are having consistent hard fails. Any temporary fails are retried with a back off automatically until a limit is reached and the message is deemed as a hard fail.

> is there a feature to goup mx servers and to send them throttled messages to avoid any blocking and to follow their guideline and not to exceed the amount of emails sent from one ip

I am not fully sure what you are looking for here but Postal has a messages-per-hour limit you can set and it has IP Pools to ensure that messages can be sent from selective IP addresses to avoid issues if someone sends a lot of bad messages through your Postal and gets your IP address blocked. 

> in terms of scaling how many emails per minute i can send from a 4 CUP and 4 RAM , SSD server ?

There is not really an answer to this because Postal can also be limited by how you have set up your database. If it is on the same server as your Postal then it will be slightly faster but you will need extra memory to deal with that. If your database is fully separate then you will need to ensure there is a good network between the two.

**Reply by z3d3m0n** [reply]
Hello,

thank you very much for your response and clarification.

regarding the  back off rulles is there any setting file where i can adjust the backoff conditions and limits ?

i will give an example regarding mx grouping  different mtas have different name for this feature but its vital for ip and domain rep 
the mailerq calls it MX Pattern Grouping and powermta MX Rollup.

as we know outlook and gmail have connection and bounce and sending limit per ip the MX Rollup or MX Grouping aggregates emails to domains with shared MX records into single routing paths. Instead of handling each domain individually, it combines them, which reduces the number of SMTP connections and optimizes delivery performance. This feature helps manage server resources more efficiently and improves deliverability by reducing load on the receiving servers.

when you have 20 organisation send to different domains but all of these domains are hoseted on gmail mx groping is essential to not exceed the limit of connections and message rate from ips.

is there an option to add custom headers ?
we would like to add one click unsubscribe links to ensure compience with google policy

thank you in advance

**Reply by willpower232** [reply]
> regarding the back off rulles is there any setting file where i can adjust the backoff conditions and limits ?

I don't believe so

> mx grouping is essential to not exceed the limit of connections and message rate from ips.

I don't think this is a feature I'm afraid. Postal is mostly aimed at transactional email rather than bulk or marketing email sending

> is there an option to add custom headers ?

Postal does not generate its own messages, apart from testing or password resets. If you want to add your own headers, you'll have to do it in whatever is sending the message. Postal does not maintain any contact or mailing lists so has no business adding an unsubscribe.

**Reply by z3d3m0n** [reply]
thank you very much for clarification 

if the user is injecting the message without using a third party marketing tool raw using SMTP details. Is there a way to  to include List-Unsubscribe-Post and List-Unsubscribe to comply with google policy even if we process on our end. 

Since we don’t own the subnet and have obtained authorization to provide email services on the condition of adhering to all policies, it is critical for us to add custom headers to every email sent via SMTP or API.It is crucial for ensuring compliance and effective email management.

**Reply by willpower232** [reply, marked as answer]
Afraid not, if the user can't put the headers in at the source then Postal can't help.

**Reply by z3d3m0n** [reply]
if we decided to give it a try and fork the git to make a programatic way in , where should we look at for adding a custom header ?

thank you
